pkg/utils/containerRestartCount.go

```diff
@@ -0,0 +1,23 @@
+package utils
+
+import (
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+func ContainerRestartCount(restartCountSumBefore int, podNs string, podLabels string, client *kubernetes.Clientset) int {
+	restartCountSumAfter := 0
+	// Getting the list of pods with given label and namespace
+	crc, err := client.CoreV1().Pods(podNs).List(metav1.ListOptions{LabelSelector: podLabels})
+	Expect(err).To(BeNil(), "fail to get the csp pods")
+	// Iterating over all pods
+	for _, podSpec := range crc.Items {
+		//Iterating over all container in a pod
+		for i := 0; i < len(podSpec.Status.ContainerStatuses); i++ {
+			restartCountSumAfter = restartCountSumAfter + int(podSpec.Status.ContainerStatuses[i].RestartCount)
+		}
+	}
+	return (restartCountSumBefore - restartCountSumAfter)
```

Comment:
Maybe good to map a container with restartCount before v/s after on individual basis. For some experiments:

- Restart on other container maybe bug
- Restart on other container is expected

Reply:
Better to make this func accept containers of type (target / unaffected)

Reply:
compareContainerRestartCount.go